frontend/packages/console-dynamic-plugin-sdk/src/extensions/yaml-templates.ts

```diff
@@ -0,0 +1,24 @@
+import { Extension } from '@console/plugin-sdk/src/typings/base';
+import { K8sKind } from 'public/module/k8s/types';
+
+export namespace ExtensionProperties {
+  export type YAMLTemplate = {
+    /** Model associated with the template. */
+    model: K8sKind;
+    /** The YAML template to use. */
+    template: string;
+    /** The name of the template. If not specified, use the `default` value. */
+    templateName?: string;
+  };
+}
+
+// Extension types
+
+export type YAMLTemplate = Extension<ExtensionProperties.YAMLTemplate> & {
+  type: 'console.yaml/template';
```

Comment:
Since this isn't a sub type of the `console.yaml` extension, it should probably as follows:
```suggestion
  type: 'console.yaml-template';
```